graylog2-web-interface/src/views/components/datatable/MessagesTable.jsx

```diff
@@ -0,0 +1,142 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+
+import { Table } from 'components/graylog';
+
+const MessagesContainer = styled.div`
+  padding-right: 13px;
+  width: 100%;
+`;
+
+const StyledTable = styled(Table)`
+  position: relative;
+  font-size: 11px;
+  margin-top: 15px;
+  margin-bottom: 60px;
+  border-collapse: collapse;
+  padding-left: 13px;
+  width: 100%;
+  word-break: break-all;
+
+  thead > tr {
+    color: #fff;
+  }
+
+  td,
+  th {
+    position: relative;
+    left: 13px;
+  }
+
+  > thead th {
+    border: 0;
+    font-size: 11px;
+    font-weight: normal;
+    background-color: #222;
+    white-space: nowrap;
+  }
+
+  tr {
+    border: 0 !important;
+  }
+
+  tbody.message-group {
+    border-top: 0;
+  }
+
+  tbody.message-group-toggled {
+    border-left: 7px solid #16ace3;
+  }
+
+  tbody.message-highlight {
+    border-left: 7px solid #8dc63f;
+  }
+
+  tr.fields-row {
+    cursor: pointer;
+  }
+
+  tr.fields-row td {
+    padding-top: 10px;
+  }
+
+  tr.message-row td {
+    border-top: 0;
+    padding-top: 0;
+    padding-bottom: 5px;
+    font-family: monospace;
+    color: #16ace3;
+  }
+
+  tr.message-row {
+    margin-bottom: 5px;
+    cursor: pointer;
+  }
+
+  tr.message-row .message-wrapper {
+    line-height: 1.5em;
+    white-space: pre-line;
+    max-height: 6em; /* show 4 lines: line-height * 4 */
+    overflow: hidden;
+  }
+
+  tr.message-row .message-wrapper:after {
+    content: "";
+    text-align: right;
+    position: absolute;
+    width: 99%;
+    left: 5px;
+    top: 4.5em;
+    height: 1.5em;
+    background: linear-gradient(to bottom, rgba(255, 255, 255, 0), rgba(255, 255, 255, 1) 95%);
+  }
+
+  tr.message-detail-row {
+    display: none;
+  }
+
+  tr.message-detail-row td {
+    padding-top: 5px;
+    border-top: 0;
+  }
+
+  tr.message-detail-row .row {
+    margin-right: 0;
+  }
+
+  tr.message-detail-row div[class*="col-"] {
+    padding-right: 0;
+  }
+
+  th i.sort-order-desc {
```

Comment:
Do we need these styles in the global styles?
https://github.com/Graylog2/graylog2-server/blob/f0bb3dc08c0b1ec64445c6eea15f04351cdd50fd/graylog2-web-interface/src/theme/GlobalThemeStyles.jsx#L776-L789